src/utils/promisifyTypeMethod.js

```diff
@@ -0,0 +1,43 @@
+import { getIsArgumentCallback } from './getIsArgumentCallback';
+
+/**
+ * Maps type of argument from scheme to method which return the argument value for original method
+ */
+const schemeArgTypeToExternalArgValueMap = {
+  value: externalArgs => externalArgs.shift(),
+  successCallback: (_, resolveCallback) => resolveCallback,
+  errorCallback: (_, __, rejectCallback) => rejectCallback,
+  objectWithCallback: (externalArg, resolveCallback, rejectCallback) => Object.assign(externalArg || {}, { onSuccess: resolveCallback, onFailure: rejectCallback })
+};
+
+/**
+ * Wraps a method of object in Promise if it is asyncronous method by scheme
+ *
+ * @param {string} methodName name of original method
+ * @param {import('../../types').Callback} methodValue original method from object
+ * @param {import('../../types').SchemeArg[]} argsFromScheme scheme of method arguments
+ * @param {import('../../types').Callback} [mapAfterCallback] mapAfter callback from scheme
+ * @returns {import('../../types').ObjectValue}
+ */
+export function promisifyTypeMethod(methodName, methodValue, argsFromScheme, mapAfterCallback = arg => arg) {
```

Comment:
Likewise, it better to have same signature:
```js
export function promisifyTypeMethod(methodValue, methodScheme)
```

Reply:
May be the name of method to write? I use it in error message.